bowtie/_core.py

```diff
@@ -538,6 +544,74 @@ def supports(self, *dialects: Dialect) -> bool:
         """
         return self.info.dialects.issuperset(dialects)
 
+    async def get_versions(self):
+        from github3 import GitHub  # type: ignore[reportMissingTypeStubs]
+        from github3.exceptions import (  # type: ignore[reportMissingTypeStubs]
+            GitHubError,
+        )
+        from github3.models import (  # type: ignore[reportMissingTypeStubs]
+            GitHubCore,
+        )
+        from github3.structs import (  # type: ignore[reportMissingTypeStubs]
+            GitHubIterator,
+        )
+        from sortedcontainers import (  # type: ignore[reportMissingTypeStubs]
+            SortedSet,
+        )
+
+        url = CONTAINER_PACKAGES_API / self.id / "versions"
+        versions = SortedSet(
+            key=lambda tag: (  # type: ignore[reportUnknownLambdaType]
+                [
+                    -int(part) if part.isdigit() else part
+                    for part in cast(str, tag).split(".")
+                ]
+            ),
+        )
+
+        try:
+            gh = GitHub(token=os.environ.get("GITHUB_TOKEN", ""))
+
+            class _TagsWrapper(GitHubCore):
```

Comment:
This is a bunch of wrapping and unwrapping, I'm not sure it does much good -- does this not work if you try simply using `GitHubCore._iter()` -- i.e. not really defining anything new, just calling that method with your parameters, which it looks like already returns a `GitHubIterator`?